pkg/datasource/opensergo/opensergo.go

```diff
@@ -0,0 +1,204 @@
+	ds.client.UnsubscribeConfig(*faultToleranceRuleSubscribeKey)
+}
+
+func (ds *OpenSergoDataSource) subscribeFlowRule() {
+	sentinelFlowRuleSubscriber := NewSentinelFlowRuleSubscriber(ds.openSergoRuleAggregator)
+	// Subscribe RateLimitStrategy
+	rlStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefRateLimitStrategy{})
+	ds.client.SubscribeConfig(*rlStrategyKey, api.WithSubscriber(sentinelFlowRuleSubscriber))
+	logging.Info("[OpenSergoDatasource] Subscribing OpenSergo RateLimitStrategy.", "namespace", ds.namespace, "app", ds.app)
+	// Subscribe ThrottlingStrategy
+	thlStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefThrottlingStrategy{})
+	ds.client.SubscribeConfig(*thlStrategyKey, api.WithSubscriber(sentinelFlowRuleSubscriber))
+	logging.Info("[OpenSergoDatasource] Subscribing OpenSergo ThrottlingStrategy.", "namespace", ds.namespace, "app", ds.app)
+}
+
+func (ds *OpenSergoDataSource) unSubscribeFlowRule() {
+	// Un-Subscribe RateLimitStrategy
+	rlStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefRateLimitStrategy{})
+	ds.client.UnsubscribeConfig(*rlStrategyKey)
+	logging.Info("[OpenSergoDatasource] un-subscribing OpenSergo RateLimitStrategy.", "namespace", ds.namespace, "app", ds.app)
+	// Un-Subscribe ThrottlingStrategy
+	thlStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefThrottlingStrategy{})
+	ds.client.UnsubscribeConfig(*thlStrategyKey)
+	logging.Info("[OpenSergoDatasource] un-subscribing OpenSergo ThrottlingStrategy.", "namespace", ds.namespace, "app", ds.app)
+	// Un-Subscribe ConcurrencyLimitStrategy
+	clStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefConcurrencyLimitStrategy{})
+	ds.client.UnsubscribeConfig(*clStrategyKey)
+	logging.Info("[OpenSergoDatasource] un-subscribing OpenSergo ConcurrencyLimitStrategy.", "namespace", ds.namespace, "app", ds.app)
+}
+
+func (ds *OpenSergoDataSource) subscribeIsolationRule() {
+	isolationRuleSubscriber := NewIsolationRuleSubscriber(ds.openSergoRuleAggregator)
+	// Subscribe ConcurrencyLimitStrategy
+	clStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefConcurrencyLimitStrategy{})
+	ds.client.SubscribeConfig(*clStrategyKey, api.WithSubscriber(isolationRuleSubscriber))
+	logging.Info("[OpenSergoDatasource] Subscribing OpenSergo ConcurrencyLimitStrategy.", "namespace", ds.namespace, "app", ds.app)
+}
+
+func (ds *OpenSergoDataSource) unSubscribeIsolationRule() {
+	// Un-Subscribe ConcurrencyLimitStrategy
+	clStrategyKey := model.NewSubscribeKey(ds.namespace, ds.app, configkind.ConfigKindRefConcurrencyLimitStrategy{})
+	ds.client.UnsubscribeConfig(*clStrategyKey)
+	logging.Info("[OpenSergoDatasource] un-subscribing OpenSergo ConcurrencyLimitStrategy.", "namespace", ds.namespace, "app", ds.app)
+}
+
+func (ds *OpenSergoDataSource) subscribeCircuitBreakerRule() {
```

Comment:
These subscribe/unsubscribe methods for Sentinel rules should be public.

Reply:
And please pay attention to error handling. Do not ignore the error here.

Reply:
> These subscribe/unsubscribe methods for Sentinel rules should be public.

i make it private because i have no idea about how to avoid re-invoking this functions. 
The subscribe functions have been invoked in `Initialize()`.  If make them public , users would invoke twice, also unsubscribe twice will be invoked twice.